### PR TITLE
Low Level Hygen configuration

### PR DESCRIPTION
**Pull request by ironBatista**
- [X] Created templates folder with hygen API calls.
- [X] Created lib folder with generator definitions.
- [X] Created documentation for hygen configuration and indexed it from the main Readme file.

### CODE REVIEW

File: hygen/lib/properties/properties.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * @returns {Array<string>}
+ */
+module.exports = function (url, args) {
+  const properties = getApiDocs(url, args);
+  const lines = [];
+
+  properties.forEach((property) => {
+    if (Array.isArray(property)) {
+      property.pop();
+      lines.push(tab.repeat(4) + "'" + property + "',");
```
Comment (mmadariaga) [inline hygen/lib/properties/properties.js:17]: property is an array so.. I don't get this

File: hygen/lib/api/api-spec.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+const getObjKeys = require('../decomposition/obj-keys');
+const mergeProperties = require('../merger/merger-properties');
+
+module.exports = function (url, args){
```
Comment (mmadariaga) [inline hygen/lib/api/api-spec.js:4]: coment types

File: hygen/lib/api/api-spec.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    let properties = [];
+    let res = request("GET", url, {
+      headers: {
+        "Content-type": "application/json",
+      },
+    });
+  
+    data = JSON.parse(res.getBody("utf8"));
+    definitions = data.definitions;
+    properties = getObjKeys(definitions[args]["properties"]);
```
Comment (mmadariaga) [inline hygen/lib/api/api-spec.js:21]: is args entity name? misleading name

File: hygen/lib/properties/properties.js (hunk 1 above)
Comment (mmadariaga) [inline hygen/lib/properties/properties.js:17]: are you concatenating an array and a string? I don't understand

File: hygen/lib/template-formatter/fmt-entity-properties.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    const properties = getApiDocs(url, args);
+    const lines = [];
+  
+    properties.forEach((property) => {
+      if (Array.isArray(property)) {
+        const enumObj = property[1];
+        lines.push(tab + "'" + property[0] + "': {");
+        lines.push(
+          tab.repeat(2) +
+            "label: _('" +
```
Comment (mmadariaga) [inline hygen/lib/template-formatter/fmt-entity-properties.js:19]: wrong indentation

File: hygen/lib/properties/properties.js (hunk 1 above)
Comment (ironBatista) [reply hygen/lib/properties/properties.js:17]: This is what happens if you call just the array as an string 
image

File: hygen/lib/api/get-api-properties.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    headers: {
+      "Content-type": "application/json",
+    },
+  });
+  let data = JSON.parse(res.getBody("utf8"));
+  let definitions = data.definitions;
+  let properties = undefined;
+
+  for (const definitionName in definitions) {
+    if (definitionName.indexOf(entityName) != -1) {
```
Comment (mmadariaga) [inline hygen/lib/api/get-api-properties.js:27]: This will match Ddi and DdiProvider

File: hygen/lib/template-formatter/fmt-entity-properties.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        "label: _('" +
+        getComposedPropertyName(propertyName) +
+        "'),"
+    );
+    if (propertyValue.enum) {
+      const enumObj = propertyValue.enum;
+      lines.push(tab.repeat(2) + "enum: {");
+      enumObj.forEach((enumValue) => {
+        lines.push(
+          tab.repeat(3) + "'" + enumValue + "' : '" + enumValue + "',"
```
Comment (mmadariaga) [inline hygen/lib/template-formatter/fmt-entity-properties.js:27]: tab.repeat(3) + "'" + enumValue + "' : '" + getComposedPropertyName(propertyName) + "',"

**Review by mmadariaga — APPROVED** [review]

**Review by mmadariaga — APPROVED** [review]

**Pull request merged**